### PR TITLE
Populate ucontext.uc_stack if using alternate stack

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vitikoo@microsoft.com>

### CODE REVIEW

File: kernel/signal.c [hunk 1]
```diff
@@ -286,6 +286,15 @@ static long _handle_one_signal(
 
         if ((action->flags & SA_SIGINFO) != 0)
         {
+            // pass stack limits if using alternate stack.
+            // dotnet runtime uses the uc_stack field to detect if its running
+            // on an alternate stack.
+            if (use_alt_stack)
+            {
+                context.uc_stack.ss_sp = altstack->ss_sp;
+                context.uc_stack.ss_sp = altstack->ss_size;
```
Comment (bodzhang) [inline kernel/signal.c]: Should the code copy ss_flags too?

Comment (vtikoo) [reply kernel/signal.c]: Added.

File: kernel/signal.c [hunk 2]
```diff
@@ -286,6 +286,16 @@ static long _handle_one_signal(
...
         if ((action->flags & SA_SIGINFO) != 0)
         {
+            // pass stack limits if using alternate stack.
+            // dotnet runtime uses the uc_stack field to detect if its running
+            // on an alternate stack.
+            if (use_alt_stack)
+            {
+                context.uc_stack.ss_sp = altstack->ss_sp;
+                context.uc_stack.ss_size = altstack->ss_size;
+                context.uc_stack.ss_flags = altstack->ss_flags;
```
Comment (bodzhang) [inline kernel/signal.c:296]: `ss_flags.SS_ONSTACK` is set on line 317. Probably should move the code block down there.

Comment (vtikoo) [reply kernel/signal.c]: Thanks for catching that. Fixed.

**Review by jxyang — APPROVED** [review]

**Pull request merged**